### DISCUSSION TITLE
question about missing bc files

### DISCUSSION THREAD

**Discussion by zjin-lcf** [AdaptiveCpp/AdaptiveCpp#635] [General]
The hipSYCL nightly for rocm was installed:
hipSYCL-base-0.8.1-20200320.x86_64
hipSYCL-0.8.1-20200825.x86_64
hipSYCL-base-rocm-nightly-0.9.1-20210516.x86_64

compiling codes show the following errors. What arguments should be used to invoke syclcc for generating code ?

syclcc -o add add.cpp -O3 --hipsycl-gpu-arch=gfx908 --hipsycl-platform=rocm
clang-9: error: no such file or directory: 'hip.amdgcn.bc'
clang-9: error: no such file or directory: 'opencl.amdgcn.bc'
clang-9: error: no such file or directory: 'ocml.amdgcn.bc'
clang-9: error: no such file or directory: 'ockl.amdgcn.bc'
clang-9: error: no such file or directory: 'oclc_finite_only_off.amdgcn.bc'
clang-9: error: no such file or directory: 'oclc_daz_opt_off.amdgcn.bc'
clang-9: error: no such file or directory: 'oclc_correctly_rounded_sqrt_on.amdgcn.bc'
clang-9: error: no such file or directory: 'oclc_unsafe_math_off.amdgcn.bc'
clang-9: error: no such file or directory: 'oclc_isa_version_908.amdgcn.bc'
clang-9: error: no such file or directory: 'oclc_wavefrontsize64_on.amdgcn.bc'

**Comment by illuhad** [comment]
hipSYCL 0.8 is deprecated. This package is only provided for compatibility reasons. Please install a recent hipSYCL (0.9.x) and base package.

The issue you are seeing is because you combine an old base package from the 0.8 series which uses clang 9 with a newer ROCm. clang 9 does not support newer versions of ROCm such as ROCm 4.x because AMD changed the directory layout, so clang doesn't find the bitcode libraries.

**Reply by zjin-lcf** [reply]
Thank you for explaining the cause.
